ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactionQueryBuilder.java

```diff
@@ -237,7 +248,11 @@ String build() {
       query.append(" temporary external");
     }
     if (operation == Operation.INSERT) {
-      query.append(" into");
+      if (compactionType.equals(CompactionType.REBALANCE)) {
+        query.append(" overwrite");
+      } else {
+        query.append(" into");
+      }
```

Comment:
Can we do this - 
`query.append(compactionType.equals(CompactionType.REBALANCE) ? "overwrite" : "append");`

Reply:
`CompactionType.REBALANCE.equals(compactionType)` would be null safe too.

Or equality operator is also fine because `CompactionType` is an enum.

Reply:
Fixed.